core/src/main/java/io/grpc/Attributes.java

```diff
@@ -58,7 +60,16 @@ private Attributes() {
   @SuppressWarnings("unchecked")
```

Comment:
Yay! Now the unchecked cast is actually safe since it can't be subverted with an incorrectly constructed key.